src/main/scala/elevate/rise/rules/algorithmic.scala

```diff
@@ -81,30 +120,77 @@ object algorithmic {
       case App(Id(), arg) => Success(app(transpose >> transpose, arg) :: e.t)
       case _              => Failure(createTransposePair)
     }
-    override def toString = "createTransposePair"
+    override def toString: String = "createTransposePair"
   }
 
-  def `_-> T >> T`: Strategy[Rise] = idAfter `;` createTransposePair
-
   def removeTransposePair: Strategy[Rise] = `T >> T -> `
   case object `T >> T -> ` extends Strategy[Rise]  {
     def apply(e: Rise): RewriteResult[Rise] = e match {
       case App(Transpose(), App(Transpose(), x)) => Success(x :: e.t)
       case _                                     => Failure(removeTransposePair)
     }
-    override def toString = "createTransposePair"
+    override def toString: String = "createTransposePair"
   }
 
   // slideSeq fusion
   import rise.OpenCL.primitives._
   import rise.OpenCL.TypedDSL._
 
-  def slideSeqFusion: Strategy[Rise] = `slideSeq(f) >> map(g) -> slideSeq(f >> g)`
-  def `slideSeq(f) >> map(g) -> slideSeq(f >> g)`: Strategy[Rise] = {
-    case expr@App(App(Map(), g), App(App(App(DepApp(DepApp(SlideSeq(rot), sz: Nat), sp: Nat), wr), f), e)) =>
+  def slideSeqFusion: Strategy[Rise] = {
+    case expr@App(App(Map(), g),
+    App(App(App(DepApp(DepApp(SlideSeq(rot), sz: Nat), sp: Nat), wr), f), e)) =>
       Success(slideSeq(rot)(sz)(sp)(wr)(typed(f) >> g)(e) :: expr.t)
-    case expr@App(App(Map(), g), App(App(App(DepApp(DepApp(DepApp(OclSlideSeq(rot), a: AddressSpace), sz: Nat), sp: Nat), wr), f), e)) =>
+    case expr@App(App(Map(), g),
+    App(App(App(DepApp(DepApp(DepApp(OclSlideSeq(rot), a: AddressSpace),
+    sz: Nat), sp: Nat), wr), f), e)) =>
       Success(oclSlideSeq(rot)(a)(sz)(sp)(wr)(typed(f) >> g)(e) :: expr.t)
     case _ => Failure(slideSeqFusion)
   }
+
+  // the inner strategies shouldn't be accessible from the outside
+  // because they might change the semantics of a program
+  case object freshLambdaIdentifier extends Strategy[Rise] {
```

Comment:
is this used to keep unique identifiers? can we just drop the unique identifiers assumption during rewriting?

Reply:
yes, I think I also added this because in the generate dot-file multple subtrees pointed to the same identifier node which was confusing